tritium-processor/src/test/resources/com/palantir/tritium/examples/ProxiedBindsParameter.java.generated

```diff
@@ -0,0 +1,74 @@
+package com.palantir.tritium.examples;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.UndeclaredThrowableException;
+import javax.annotation.processing.Generated;
+
+@Generated("com.palantir.tritium.processor.ProxyAnnotationProcessor")
+public final class ProxiedBindsParameter implements BindsParameter {
+    private static final Method EQUALS;
+
+    private static final Method GET;
+
+    private static final Method TO_STRING;
+
+    static {
+        try {
+            EQUALS = BindsParameter.class.getMethod("equals", Object.class);
+            GET = BindsParameter.class.getMethod("get");
+            TO_STRING = BindsParameter.class.getMethod("toString");
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final InvocationHandler handler;
+
+    private ProxiedBindsParameter(InvocationHandler handler) {
+        this.handler = handler;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean equals(Object obj) {
+        try {
+            Object returned = this.handler.invoke(this, EQUALS, new Object[]{obj});
+            return (boolean) returned;
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CharSequence get() {
+        try {
+            Object returned = this.handler.invoke(this, GET, new Object[]{});
+            return (CharSequence) returned;
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public String toString() {
+        try {
+            Object returned = this.handler.invoke(this, TO_STRING, new Object[]{});
+            return (String) returned;
+        } catch (Error | RuntimeException throwable) {
+            throw throwable;
+        } catch (Throwable throwable) {
+            throw new UndeclaredThrowableException(throwable);
+        }
+    }
+
+    public static BindsParameter of(InvocationHandler handler) {
```

Comment:
So the flow would be:

1. Annotate your MyService interface with @Proxy.
2. In your code call ProxiedMyService.of(InvocationHandler).
3. Profit!

So this would replace all the usages of java.lang.reflect.Proxy.newProxyInstance. AbstractInvocationHandler usages should JUST work? https://github.com/google/guava/blob/master/guava/src/com/google/common/reflect/AbstractInvocationHandler.java#L44

Reply:
Yep, that's the plan!